### ISSUE TITLE
Adding tracking to pose estimation

### ISSUE THREAD

**Issue by sard0r** [ultralytics/ultralytics#4486]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Currently I have an action recognition model which includes yolov8 detection, sort tracking, fastpose pose estimation and action recognition model. I found that it is very resource consuming compared to using yolov8 pose estimation model.

I tested yolov8 pose estimation model and found that it automatically detects and estimates the keypoints of people. What I want to ask is can we add tracking model for example bytterack to track humans and objects in the frame. 

Thank you

**Comment by glenn-jocher** [comment]
@sard0r hello, thanks for reaching out with your question!

Yes, it's certainly possible to integrate a tracking model like Deep SORT or ByteTrack with the YOLOv8 for better efficiency. Start by obtaining the bounding box outputs from the YOLOv8 detector, which you could then feed into your chosen tracking algorithm. This will allow the tracking algorithm to track those detected objects across frames.

In regards to pose estimation, as you mentioned, YOLOv8 already provides keypoint estimation for detected humans. If you want to use this feature in combination with a tracker, it's again a matter of integrating the separate components. The keypoints detected by YOLOv8 could be used as input for the action recognition model or any other module that could use this information.

I hope this clarifies your question. Please let us know if you have any further queries. Enjoy coding with YOLOv8!

**Comment by sard0r** [comment]
@glenn-jocher Thank you for your answer. For more I wanted to ask if it is possible to detect hand and face keypoints using yolov8.

**Comment by glenn-jocher** [comment]
@sard0r hello, thank you for your follow-up question.

The capabilities of keypoint detection within YOLOv8 largely depend on the training data and the annotation format. If you want to detect hand and face keypoints, you would need to have a dataset that specifically includes these as annotated keypoints.

The YOLOv8 model in its current state has been trained on data that includes keypoints for full human body poses, not specifically for hands and faces. So, for detecting hand and face keypoints, you would need to train a YOLOv8 model with an appropriate dataset that includes these keypoints.

In summary, it is possible to detect hand and face keypoints with YOLOv8 provided you have suitable training data. I hope this clarifies your question, and please feel free to ask if you have more!

**Comment by dizcza** [comment]
Can someone point us to a working example utilizing yolov8 pose model and a (third-party) keypoints tracker on top? Instead of saying "it can be used with any tracker".

@sard0r did you find a working example?

Thanks.

**Comment by glenn-jocher** [comment]
You can run ie 'yolo track model=yolov8n-pose.yaml source=path/to/video'

**Comment by dizcza** [comment]
True, I haven't looked into the tracker folder: there are two good-written trackers in python.

Has anyone seen Yolo pose tracker in C++? I cannot find any example.

**Comment by glenn-jocher** [comment]
@dizcza hello, thanks for your interest in incorporating a tracker with the YOLOv8 pose model. While there are Python implementations available, I understand you are specifically looking for a C++ example.

Currently, there isn't a predefined or officially supported YOLOv8 pose tracker in C++. However, it's certainly possible to develop one. A C++ implementation would involve extracting the bounding box predictions from the YOLOv8 pose model and then feeding these into your tracking algorithm of choice. You can use any C++ based object tracking library or algorithm.

Remember, the pose estimation model not only locates persons but also provides the keypoints. These keypoints can be used as an additional source of information by your tracking model.

Please don't hesitate to ask if you have more questions or need further clarification. Happy coding with YOLOv8!
